array.c

```diff
@@ -3521,6 +3521,120 @@ sort_2(const void *ap, const void *bp, void *dummy)
     return n;
 }
 
+/* Radix_Sort */
+
+typedef struct rsort_double_t { uint64_t z; VALUE v; } rsort_double_t;
+
+static int rsort_double(VALUE *const p, const long l) {
+
+    rsort_double_t *const _r = malloc(l * 2 * sizeof(rsort_double_t)),
+                          *a = _r, *b = a + l, *t;
+    if (_r == NULL) { return 1; } _Bool is_unordered = 0;
+
+    uint64_t F[8][256] = {{0}}, prev = 0;
+
+    for (long i = 0; i < l; i++) {
+        union { double d; uint64_t z; } u;
+        if (!RB_FLOAT_TYPE_P(p[i]) || isnan(u.d = rb_float_value(p[i]))) {
+            free(_r);
+            return 1;
+        }
+        u.z ^= ((int64_t)(u.z) >> 63) | ((uint64_t)1 << 63);
+        is_unordered |= prev > u.z, prev = u.z;
+        for (int i = 0; i < 8; i++)
+            F[i][(uint8_t)(u.z >> i * 8)]++;
+        a[i] = (rsort_double_t){ u.z, p[i] };
+    }
+
+    if (!is_unordered) { free(a); return 0; }
+
+    uint64_t skip[8] = {0};
+
+    for (int i = 0; i < 8; i++) {
+        uint64_t x = 0, t, *o = F[i];
+        for (int j = 0; j < 256; j++) {
+            if ((t = o[j]) == (uint64_t)l) {
+                skip[i] = 1;
+                break;
+            }
+            x = (o[j] = x) + t;
+        }
+    }
+
+    for (int i = 0; i < 8; i++) {
+        if (skip[i]) continue;
+        uint64_t *o = F[i];
+        for (rsort_double_t *p = a, *const P = p + l; p < P; p++)
+            b[o[(uint8_t)(p->z >> i * 8)]++] = *p;
+        t = a, a = b, b = t;
+    }
+
+    for (long i = 0; i < l; i++)
+        p[i] = a[i].v;
+
+    free(_r);
+    return 0;
+}
+
+static int rsort(void *const _p, const long l) {
+
+    if (SIZEOF_VALUE != 8) return 1;
```

Comment:
Why not supporting `VALUE` other than 64bit?
Replacing `uint64_t` with `VALUE` can relax this restriction?